src/resources/Microsoft.Authorization/policyDefinitions/Deny-VMUnmanagedDisk.json

```diff
@@ -0,0 +1,67 @@
+{
+    "name": "Deny-VM-with-unmanaged-disk",
+    "apiVersion": "2021-06-01",
+    "scope": "null",
+    "properties": {
+        "policyType": "Custom",
+        "mode": "all",
+        "displayName": "Deny virtual machines and virtual machine scalesets not using managed disk",
+        "description": "Deny virtual machines not using managed disk. It checks the managedDisk property on virtual machine OS, and Datadisk fields.",
+        "metadata": {
+            "version": "1.0.0.0",
+            "category": "Compute",
+            "source": "https://github.com/Azure/Enterprise-Scale/",
+            "alzCloudEnvironments": [
+                "AzureCloud"
+            ]
+        },
+        "parameters": {
+            "effectType": {
+                "type": "string",
+                "defaultValue": "Deny",
+                "allowedValues": [
+                    "Audit",
+                    "Deny",
+                    "Disabled"
+                ],
+                "metadata": {
+                    "displayName": "Effect",
+                    "description": "Enable or disable the execution of the policy"
+                }
+            }
+        },
+        "policyRule": {
+            "if": {
+                "anyOf": [
+                    {
+                        "allOf": [
+                            {
+                                "field": "type",
+                                "equals": "Microsoft.Compute/virtualMachines"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachines/storageProfile.osDisk.managedDisk",
+                                "exists": "false"
+                            }
+                        ]
+                    },
+                    {
+                        "allOf": [
+                            {
+                                "field": "type",
+                                "equals": "Microsoft.Compute/virtualMachineScaleSets"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachineScaleSets/virtualMachineProfile.storageProfile.osDisk.managedDisk",
+                                "exists": "false"
+                            }
+                        ]
+                    }
+                ]
+            },
+            "then": {
+                "effect": "[parameters('effectType')]"
```

Comment:
```suggestion
                "effect": "[[parameters('effectType')]"
```

Reply:
I missed the escaping of the formula. THanks